daemon/cmd/api_handlers.go

```diff
@@ -0,0 +1,213 @@
+package cmd
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/cilium/cilium/api/v1/server/restapi/daemon"
+	"github.com/cilium/cilium/api/v1/server/restapi/endpoint"
+	"github.com/cilium/cilium/api/v1/server/restapi/ipam"
+	"github.com/cilium/cilium/api/v1/server/restapi/metrics"
+	"github.com/cilium/cilium/api/v1/server/restapi/policy"
+	"github.com/cilium/cilium/api/v1/server/restapi/prefilter"
+	"github.com/cilium/cilium/api/v1/server/restapi/recorder"
+	"github.com/cilium/cilium/api/v1/server/restapi/service"
+	"github.com/cilium/cilium/pkg/api"
+	datapathOption "github.com/cilium/cilium/pkg/datapath/option"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/promise"
+	"github.com/go-openapi/runtime/middleware"
+)
+
+type handlersOut struct {
+	cell.Out
+
+	DaemonGetCgroupDumpMetadataHandler daemon.GetCgroupDumpMetadataHandler `optional:"true"`
```

Comment:
Right yeah this needs to be documented. Though I did just shoot myself in the foot with it.. I had the endpoint handler depend on ratelimiter that I hadn't provided and it just ignored it rather than complained about it (e.g. the optionality was infectious).. so might be better that it stays undocumented for now :D

I've removed the optional from this for now.